### PR TITLE
refactor Dockerfile

### PR DESCRIPTION
**Pull request by takasehideki**
This PR mainly contributes to refactoring Dockerfile, and adjusting the official ROS 2 installation steps.
One big change is to add `apt upgrade` before installing the ROS 2 package because it is recommended as "Warning" label in https://docs.ros.org/en/humble/Installation/Ubuntu-Install-Debians.html#install-ros-2-packages

### CODE REVIEW

**Comment by takasehideki** [comment]
@pojiro could you please review this if you have time?

File: Dockerfile [hunk 1]
```diff
@@ -1,67 +1,83 @@
-### FIXME according to the target version
-# base image, ARG is overridable by --build-arg
-ARG BASE_IMAGE=hexpm/elixir:1.13.4-erlang-25.0.3-ubuntu-focal-20211006
+# Setup the target version
+## base image, ARG is overridable by --build-arg
+ARG BASE_IMAGE=hexpm/elixir:1.15.5-erlang-26.0.2-ubuntu-jammy-20230126
```
Comment (pojiro) [inline Dockerfile:3]:
```suggestion
ARG BASE_IMAGE=hexpm/elixir:1.15.7-erlang-26.2.3-ubuntu-jammy-20240125
```

もしかしたら、 [bump latest branch](https://github.com/rclex/rclex_docker/compare/main...bump_latest) と同じ version にしたい？（ニュートラルな提案

**Review by pojiro — APPROVED** [review]
I left one trivial comment, the decision is up to you.
Any other things are LGTM!

**Comment by takasehideki** [comment]
@pojiro thank you for reviewing!
はい，ご推察のとおりです．いったんこれは merge してから `bump_latest` にて更新予定です！

**Pull request merged**